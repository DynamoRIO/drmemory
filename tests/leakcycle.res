# **********************************************************
# Copyright (c) 2011-2017 Google, Inc.  All rights reserved.
# **********************************************************
#
# Dr. Memory: the memory debugger
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation;
# version 2.1 of the License, and no later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
%if X32
Error #1: LEAK 4 direct bytes + 4 indirect bytes
%endif
%if X64
Error #1: LEAK 8 direct bytes + 8 indirect bytes
%endif
# Either allocation site could be the root, it depends which had the lower
# memory address.
%ANYLINE
leakcycle.cpp:60
leakcycle.cpp:68
%ENDANYLINE
