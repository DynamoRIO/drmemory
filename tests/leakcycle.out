# **********************************************************
# Copyright (c) 2011-2017 Google, Inc.  All rights reserved.
# **********************************************************
#
# Dr. Memory: the memory debugger
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation;
# version 2.1 of the License, and no later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
done
~~Dr.M~~ ERRORS FOUND:
~~Dr.M~~       0 unique,     0 total unaddressable access(es)
~~Dr.M~~       0 unique,     0 total uninitialized access(es)
~~Dr.M~~       0 unique,     0 total invalid heap argument(s)
~~Dr.M~~       0 unique,     0 total warning(s)
%if X32
~~Dr.M~~       1 unique,     1 total,      8 byte(s) of leak(s)
%endif
%if X64
~~Dr.M~~       1 unique,     1 total,     16 byte(s) of leak(s)
%endif
~~Dr.M~~       0 unique,     0 total,      0 byte(s) of possible leak(s)
